lib/forspell/reporter.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require 'fileutils'
+require 'pastel'
+require 'logger'
+require 'json'
+
+module Forspell
+  class Reporter
+    SUCCESS_CODE = 0
+    ERROR_CODE = 1
+    FORMAT = '%<file>s:%<line>i: %<text>s (suggestions: %<suggestions>s)'
+
+    def initialize(logfile:,
+                   verbose:,
+                   format:,
+                   group:)
+
+      FileUtils.touch(logfile) if logfile.is_a?(String)
+      @logger = Logger.new(logfile || STDOUT)
+      @logger.level = verbose ? Logger::INFO : Logger::WARN
+
+      @logger.formatter = proc do |_severity, _datetime, _progname, msg|
+        "#{msg}\n"
+      end
+
+      @format = group ? 'group' : format
```

Comment:
If that's the only thing `group` parameter does, maybe it is better to do in `CLI` (and, well... In fact it is different format, like `dictionary`, not "readable format + group flag")

Reply:
Yep, just wanted it to be a separate option, as we discussed.

Reply:
Yeah, it could/should be separate from "option parser" point of view, but for reporter it is all the same -- just one more format.